src/overlays/actors/ovl_En_Nb/z_en_nb.c

```diff
@@ -1,15 +1,1553 @@
+    CsCmdActorAction* csCmdNPCAction;
+
+    if ((globalCtx->csCtx.state != 0) &&
+        (csCmdNPCAction = globalCtx->csCtx.npcActions[npcActionIdx], csCmdNPCAction != NULL) &&
+        (csCmdNPCAction->action != arg2)) {
+        return true;
+    }
+    return false;
+}
+
+void EnNb_SetInitialCsPosRot(EnNb* this, GlobalContext* globalCtx, s32 npcActionIdx) {
+    CsCmdActorAction* csCmdNPCAction = EnNb_GetNpcCsAction(globalCtx, npcActionIdx);
+    s16 newRotY;
+    Actor* thisx = &this->actor;
+
+    if (csCmdNPCAction != NULL) {
+        thisx->posRot.pos.x = csCmdNPCAction->startPos.x;
+        thisx->posRot.pos.y = csCmdNPCAction->startPos.y;
+        thisx->posRot.pos.z = csCmdNPCAction->startPos.z;
+        newRotY = csCmdNPCAction->rot.y;
+        thisx->shape.rot.y = newRotY;
+        thisx->posRot.rot.y = newRotY;
+    }
+}
+
+void EnNb_SetCurrentAnim(EnNb* this, AnimationHeader* animation, u8 mode, f32 transitionRate, s32 arg4) {
+    f32 frameCount = SkelAnime_GetFrameCount(&animation->genericHeader);
+    f32 playbackSpeed;
+    f32 unk0;
+    f32 fc;
+
+    if (!arg4) {
+        unk0 = 0.0f;
+        fc = frameCount;
+        playbackSpeed = 1.0f;
+    } else {
+        unk0 = frameCount;
+        fc = 0.0f;
+        playbackSpeed = -1.0f;
+    }
+
+    SkelAnime_ChangeAnim(&this->skelAnime, animation, playbackSpeed, unk0, fc, mode, transitionRate);
+}
+
+void EnNb_SetChamberAnim(EnNb* this, GlobalContext* globalCtx) {
+    EnNb_SetCurrentAnim(this, &D_06009694, 0, 0, 0);
+    this->actor.shape.unk_08 = -10000.0f;
+}
+
+void EnNb_SpawnBlueWarp(EnNb* this, GlobalContext* globalCtx) {
+    f32 posX = this->actor.posRot.pos.x;
+    f32 posY = this->actor.posRot.pos.y;
+    f32 posZ = this->actor.posRot.pos.z;
+
+    Actor_SpawnAsChild(&globalCtx->actorCtx, &this->actor, globalCtx, ACTOR_DOOR_WARP1, posX, posY, posZ, 0, 0, 0, 2);
+}
+
+void EnNb_GiveMedallion(EnNb* this, GlobalContext* globalCtx) {
+    Player* player = PLAYER;
+    f32 posX = player->actor.posRot.pos.x;
+    f32 posY = player->actor.posRot.pos.y + 50.0f;
+    f32 posZ = player->actor.posRot.pos.z;
+
+    Actor_SpawnAsChild(&globalCtx->actorCtx, &this->actor, globalCtx, ACTOR_DEMO_EFFECT, posX, posY, posZ, 0, 0, 0,
+                       0xC);
+    Item_Give(globalCtx, ITEM_MEDALLION_SPIRIT);
+}
+
+void EnNb_ComeUpImpl(EnNb* this, GlobalContext* globalCtx) {
+    this->actor.shape.unk_08 += 83.333336f;
+}
+
+void EnNb_SetupChamberCsImpl(EnNb* this, GlobalContext* globalCtx) {
+    s32 pad[2];
+    Player* player;
+    s16 yaw;
+
+    if ((gSaveContext.chamberCutsceneNum == 3) && (gSaveContext.sceneSetupIndex < 4)) {
+        player = PLAYER;
+        this->action = NB_CHAMBER_UNDERGROUND;
+        globalCtx->csCtx.segment = &D_80AB431C;
+        gSaveContext.cutsceneTrigger = 2;
+        Item_Give(globalCtx, ITEM_MEDALLION_SPIRIT);
+        yaw = this->actor.posRot.rot.y + 0x8000;
+        player->actor.shape.rot.y = yaw;
+        player->actor.posRot.rot.y = yaw;
```

Comment:
```suggestion
        player->actor.posRot.rot.y = player->actor.shape.rot.y = this->actor.posRot.rot.y + 0x8000;
```

Reply:
i tried this earlier and no luck for some reason, will try again incase i did it wrong